tests/test_course.py

```diff
@@ -213,3 +213,109 @@ def test_view(self, client_student):
         assert json['data']['TAs'][0]['username'] == 'admin'
         assert json['data']['teacher']['username'] == 'admin'
         assert json['data']['studentNicknames'] == {'student': 'noobs'}
+
+
+class TestCourseGrade(BaseTester):
+    '''Test grading feature in courses
+    '''
+    def test_add_score(self, client_admin):
+        # add scores
+
+        # create a course
+        client_admin.post('/course',
+                          json={
+                              'course': 'math',
+                              'teacher': 'admin'
+                          })
+
+        # add a student
+        client_admin.put('/course/math',
+                         json={
+                             'TAs': ['admin'],
+                             'studentNicknames': {
+                                 'student': 'noobs'
+                             }
+                         })
+
+        rv = client_admin.post('/course/math/grade/student',
+                               json={
+                                   'title': 'exam',
+                                   'content': 'hard',
+                                   'score': 'A+',
+                               })
+
+        assert rv.status_code == 200
+
+        rv = client_admin.post('/course/math/grade/student',
+                               json={
+                                   'title': 'exam2',
+                                   'content': 'easy',
+                                   'score': 'F',
+                               })
+
+        assert rv.status_code == 200
+
+    def test_add_existed_score(self, client_admin):
+        # add an existed score
+        rv = client_admin.post('/course/math/grade/student',
+                               json={
+                                   'title': 'exam',
+                                   'content': '?',
+                                   'score': 'B',
+                               })
+
+        assert rv.status_code == 400
+        json = rv.get_json()
+        assert json['message'] == 'This title is taken.'
+
+    def test_modify_score(self, client_admin):
+        # modify a score
+        rv = client_admin.put('/course/math/grade/student',
+                              json={
+                                  'title': 'exam2',
+                                  'newTitle': 'exam2 (edit)',
+                                  'content': 'easy',
+                                  'score': 'E',
+                              })
+
+        assert rv.status_code == 200
```

Comment:
how about add a check for score data here?

Reply:
There's a check at the bottom, which is testing GET